herddb-core/src/main/java/herddb/file/FileDataStorageManager.java

```diff
@@ -896,24 +897,18 @@ public void writePage(String tableSpace, String tableName, long pageId, Collecti
     }
 
     private static long writeIndexPage(DataWriter writer, ManagedFile file, OutputStream stream) throws IOException {
-        long size;
-        try (XXHash64Utils.HashingOutputStream hash = new XXHash64Utils.HashingOutputStream(stream);
-                ExtendedDataOutputStream dataOutput = new ExtendedDataOutputStream(hash)) {
-
-            dataOutput.writeVLong(1); // version
-            dataOutput.writeVLong(0); // flags for future implementations
-
-            size = hash.size();
-            writer.write(dataOutput);
-            size = hash.size() - size;
-
-            // footer
-            dataOutput.writeLong(hash.hash());
-            dataOutput.flush();
-            file.sync();
-
-            return size;
-        }
+        VisibleByteArrayOutputStream oo = new VisibleByteArrayOutputStream(1024 * 1024);
```

Comment:
write in memory

Reply:
This is a full page "double" buffer. We could try to recycle it.